programs/survival/compass.sc

```diff
@@ -0,0 +1,611 @@
+    );
+);
+
+dontSpawnWithEffect(name, player) -> (
+    if (player() ~ 'permission_level' > 1,
+        global_toKeepEffect = filter(global_toKeepEffect,
+            _: 0 != name || _: 3 != player;
+        );
+        toKeepEffect = map(global_toKeepEffect, l(_: 0, str(_: 1), str(_: 2), _: 3));
+        delete_file('toKeepEffect', 'nbt');
+        write_file('toKeepEffect', 'nbt', encode_nbt(toKeepEffect));
+        print(player(), format('l Done'));
+    ,
+        print(player(), format('l Insufficient permissions', 'ei   Requires at least permission level 2'));
+    );
+);
+
+timeToTrack(val) -> (
+    if (player() ~ 'permission_level' > 1,
+        if (type(val) == 'string',
+            global_timeToTrack = val;
+        ,
+            if (global_timeToTrack == 'onUse',
+                schedule(0, 'changeCompassPos');
+            );
+            global_timeToTrack = round(val * 20);
+        );
+
+        system_variable_set('compass:timeToTrack', global_timeToTrack);
+
+        write_file('timeToTrack', 'nbt', encode_nbt(global_timeToTrack));
+        print(player(), format('l Done'));
+    ,
+        print(player(), format('l Insufficient permissions', 'ei   Requires at least permission level 2'));
+    );
+);
+
+logOnTrack(val) -> (
+    if (player() ~ 'permission_level' > 1,
+        global_logOnTrack = val;
+
+        write_file('logOnTrack', 'nbt', encode_nbt(global_logOnTrack));
+        print(player(), format('l Done'));
+    ,
+        print(player(), format('l Insufficient permissions', 'ei   Requires at least permission level 2'));
+    );
+);
+
+maxCompasses(val) -> (
+    if (player() ~ 'permission_level' > 1,
+        global_maxCompasses = val;
+
+        write_file('maxCompasses', 'nbt', encode_nbt(global_maxCompasses));
+        print(player(), format('l Done'));
+    ,
+        print(player(), format('l Insufficient permissions', 'ei   Requires at least permission level 2'));
+    );
+);
+
+noPortalBreaking(val) -> (
+    if (player() ~ 'permission_level' > 1,
+        global_noPortalBreaking = val;
+        delete_file('noPortalBreaking', 'nbt');
+        write_file('noPortalBreaking', 'nbt', encode_nbt(global_noPortalBreaking));
+        print(player(), format('l Done'));
+    ,
+        print(player(), format('l Insufficient permissions', 'ei   Requires at least permission level 2'));
+    );
+);
+
+maxNetherTravel(val) -> (
+    if (player() ~ 'permission_level' > 1,
+        global_maxNetherTravel = floor(val/8);
+
+        write_file('maxNetherTravel', 'nbt', encode_nbt(global_maxNetherTravel));
+        print(player(), format('l Done'));
+    ,
+        print(player(), format('l Insufficient permissions', 'ei   Requires at least permission level 2'));
+    );
+);
+
+includes(list, v) -> {
+    for (list,
+        if (_ == v,
+            return(true);
+        );
+    );
+
+    return(false);
+};
+
+distance(v1, v2) -> (
+    sqrt(reduce(v1, _a+(_-(v2: _i))^2, 0));
```

Comment:
suggestion for a cleaner (and possibly faster) distance function
```
sqrt(reduce(v1 - v2, _a + _*_, 0))
```

Reply:
Good idea, I totally forgot you could do that